### DISCUSSION TITLE
Отслеживание статистики

### DISCUSSION THREAD

**Discussion by Kor-Angar** [dosinabox/g2nr_unofficial_update#438] [General]
Можно дать Ищущим какой нибудь уникальный предмет что бы их можно было посчитать мне при очередном прохождение. например какой нибудь амулет который ничего не стоит.

**Comment by dosinabox** [comment]
Была у меня идея сделать книгу с разной статистикой.
- убито монстров
- убито ищущих
- убито людей
- избито людей
- ещё что-нибудь

Идеи?)

**Comment by Kor-Angar** [comment]
> Была у меня идея сделать книгу с разной статистикой.

наверное можно :-) в журнал?

**Comment by dosinabox** [comment]
В журнал не подходит. Вся информация попавшая туда уже никогда не меняется. Будет книга, доступная через консоль.

**Comment by Kor-Angar** [comment]
Тогда лучше начать с людей ящеров, элитных орков и Ищущих :-) они генерятся в 4 главах и хочется проверить не упустили кого:-)

**Comment by dosinabox** [comment]
> элитных орков

Должны ли относиться к ним предводители и полковники? Я думаю, что нет.

**Comment by dosinabox** [comment]
![Безымянный](https://user-images.githubusercontent.com/39790055/66257977-4cc4d200-e7a8-11e9-9cf6-125dee123e51.jpg)

**Comment by Kor-Angar** [comment]
А можно куда нибудь записать ягодки гоблина которые поднимаются из палов? Для статистики.

Касательно командиров орков: если они будут записаны отдельно, то можно будет понять всех ли убил герой или остались еще.

А записи можно разделить на локации? Столько то в минентале, столько то в Хориносе, столько то в Ярике? 

Если записи разделить на локации тогда коммандиров можно считать с элитнымии орками. Так как в Хориносе всего один элитный орк.

**Comment by dosinabox** [comment]
> А можно куда нибудь записать ягодки гоблина которые поднимаются из палов? Для статистики.

Не представляю, как это можно сделать.

> А записи можно разделить на локации? Столько то в минентале, столько то в Хориносе, столько то в Ярике?

Можно, но нужно подумать как это будет выглядеть в книге и хватит ли на всё места.

**Comment by Kor-Angar** [comment]
> Не представляю, как это можно сделать.

можно тогда в набор у паладина где идёт ягодка гоблина, добавить какую нибудь уникальную безделушку. например серебряное кольцо с надписью "Иннос, распространи свою длань над слугами своими"

**Comment by Kor-Angar** [comment]
> Можно, но нужно подумать как это будет выглядеть в книге и хватит ли на всё места.

можно начать собирать, а в 5 главе посмотреть  когда инфа соберётся и затем её можно будет сгруппировать

**Comment by dosinabox** [comment]
добавлен счетчик Ватраса MadKillerCount (убито невинных людей)

**Comment by Kor-Angar** [comment]
С какой главы считает?

**Comment by dosinabox** [comment]
С первой. Ватрас начинает обращать на это внимание в пятой.

**Comment by Kor-Angar** [comment]
А за пожертвование обнуление происходит? Это механика по моему в игре была сломана-- Отправлено из мобильного приложения Яндекс.Почты

**Comment by dosinabox** [comment]
> А за пожертвование обнуление происходит?

Нет. Даже не знаю, откуда такая мысль появилась в народе. Кто-то пустил утку? Димус пытался сделать такую систему в обновлении? Что-то подобное реализовано в мод-фиксе на оригинальную Г2? Сама система с Ватрасом появилась в аддоне, но механики замаливания убийств в игре просто нет.

**Comment by Kor-Angar** [comment]
Мысль такая следует из диалога с Ватросом. Он говорить что за пожертвование списываются грехи или что то такое. Запости сюда диалог плиз я честно говоря не помню что именно онименно он говорит :-)

**Comment by dosinabox** [comment]
```
func void DIA_Addon_Vatras_KillerWarning_Info()
{
	AI_Output(self,other,"DIA_Addon_Vatras_KillerWarning_ADD_05_00");	//Я слышал про тебя.
	AI_Output(self,other,"DIA_Addon_Vatras_KillerWarning_ADD_05_01");	//Говорят, что ты замешан в убийстве невинных людей.
	AI_Output(self,other,"DIA_Addon_Vatras_KillerWarning_ADD_05_02");	//Если это так, то я должен предупредить тебя, сынок.
	AI_Output(self,other,"DIA_Addon_Vatras_KillerWarning_ADD_05_03");	//Не стоит нарушать баланс сил, ибо в противном случае тебе придется столкнуться с последствиями своих поступков.
	AI_Output(self,other,"DIA_Addon_Vatras_KillerWarning_ADD_05_04");	//Чем я могу помочь тебе?
};
```

```
func void B_LastWarningVatras()
{
	AI_Output(self,other,"DIA_Addon_Vatras_LastWarning_ADD_05_00");	//Негодяй, что ты сделал?
	AI_Output(self,other,"DIA_Addon_Vatras_LastWarning_ADD_05_01");	//Орден Воды много раз предупреждал тебя об опасности попасться в сети Зла.
	AI_Output(self,other,"DIA_Addon_Vatras_LastWarning_ADD_05_02");	//Тем не менее, я узнал, что ты ушел с пути всесохраняющего равновесия.
	AI_Output(self,other,"DIA_Addon_Vatras_LastWarning_ADD_05_03");	//Ты убивал и грабил всех и вся. Ты виновен во многих преступлениях.
	AI_Output(self,other,"DIA_Addon_Vatras_LastWarning_ADD_05_04");	//Множество невинных людей погибли от ТВОЕЙ руки.
	AI_Output(self,other,"DIA_Addon_Vatras_LastWarning_ADD_05_05");	//Теперь ты - союзник сил Тьмы.
};
```

```
func void B_VatrasPissedOff()
{
	AI_Output(self,other,"DIA_Addon_Vatras_PissedOffPerm_Add_05_00");	//Отныне ты не можешь рассчитывать на нашу помощь.
	AI_Output(self,other,"DIA_Addon_Vatras_PissedOffPerm_Add_05_01");	//Убирайся. Для меня ты теперь пария.
};
```

```
func void DIA_Addon_Vatras_LastWarning_Arsch()
{
	AI_Output(other,self,"DIA_Addon_Vatras_LastWarning_Arsch_ADD_15_00");	//Отвали.
	AI_Output(self,other,"DIA_Addon_Vatras_LastWarning_Arsch_ADD_05_00");	//Я не слышу в твоих словах ни тени раскаяния.
	AI_Output(self,other,"DIA_Addon_Vatras_LastWarning_Arsch_ADD_05_01");	//Ты не оставляешь мне другого выбора.
	Info_ClearChoices(DIA_Addon_Vatras_LastWarning);
	B_VatrasPissedOff();
};
```

```
func void DIA_Addon_Vatras_LastWarning_Reue()
{
	AI_Output(other,self,"DIA_Addon_Vatras_LastWarning_Reue_ADD_15_00");	//Мне очень жаль. Я не понимал, что делаю.
	AI_Output(self,other,"DIA_Addon_Vatras_LastWarning_Reue_ADD_05_00");	//Я буду молиться за тебя и просить богов, чтобы они вернули тебе рассудок.
	AI_Output(self,other,"DIA_Addon_Vatras_LastWarning_Reue_ADD_05_01");	//Горе тебе, если я услышу хотя бы об одном убийстве, к которому ты приложил руку.
	Info_ClearChoices(DIA_Addon_Vatras_LastWarning);
	VatrasMadKillerCount = MadKillerCount;
};
```

```
func void DIA_Addon_Vatras_PissedOff_Info()
{
	AI_Output(self,other,"DIA_Addon_Vatras_PissedOff_ADD_05_00");	//Твои слова так же лживы, как и твои деяния.
	AI_Output(self,other,"DIA_Addon_Vatras_PissedOff_ADD_05_01");	//Ты никак не хочешь прекращать бездумные убийства.
	AI_Output(self,other,"DIA_Addon_Vatras_PissedOff_ADD_05_02");	//Ты не оставляешь мне другого выбора...
	B_VatrasPissedOff();
};
```

**Comment by dosinabox** [comment]
> Он говорить что за пожертвование списываются грехи или что то такое.

Видимо, это:

```
func void DIA_Vatras_Spende_Info()
{
	AI_Output(other,self,"DIA_Vatras_Spende_15_00");	//Я хочу сделать пожертвование Аданосу!
	AI_Output(self,other,"DIA_Vatras_Spende_05_01");	//Пожертвование церкви Аданоса снимет часть грехов, которые ты мог совершить, сын мой.
	AI_Output(self,other,"DIA_Vatras_Spende_05_02");	//Сколько ты можешь пожертвовать?
	Info_ClearChoices(DIA_Vatras_Spende);
	Info_AddChoice(DIA_Vatras_Spende,"Мне сейчас нечем поделиться...",DIA_Vatras_Spende_BACK);
	if(Npc_HasItems(other,ItMi_Gold) >= 50)
	{
		Info_AddChoice(DIA_Vatras_Spende,"У меня есть 50 золотых монет...",DIA_Vatras_Spende_50);
	};
	if(Npc_HasItems(other,ItMi_Gold) >= 100)
	{
		Info_AddChoice(DIA_Vatras_Spende,"У меня есть 100 золотых монет...",DIA_Vatras_Spende_100);
	};
};
```

**Comment by Kor-Angar** [comment]
Это оно 😀 сейчас я как понимаю пожертвования ни на что не влияют :(

**Comment by dosinabox** [comment]
Несколько дополнений:

![Screenshot_2](https://user-images.githubusercontent.com/39790055/75606778-53f6ce80-5b01-11ea-875b-9f1f68424a74.jpg)

**Comment by Kor-Angar** [comment]
Тогда можно добавить и Ягоды Гоблина и Драконий корень. 14:40, 29 февраля 2020 г., dosinabox <notifications@github.com>:Несколько дополнений:


—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.
-- Отправлено из мобильного приложения Яндекс.Почты

**Comment by dosinabox** [comment]
А как их считать, если они будут потрачены на зелья?

**Comment by Kor-Angar** [comment]
Считать те которые герой съел. Как яблоки и грибы.

**Comment by dosinabox** [comment]
добавлены ягоды и корни

**Comment by Kor-Angar** [comment]
может туда же молитвы добавить?

**Comment by dosinabox** [comment]
что именно?

**Comment by Kor-Angar** [comment]
Молитвы Белиара
Получено золото столько то
Потрачено жизней
Потрачено Маны

Молитвы Инноса
Сила - столько то
Ловкость - столько то
Мана - столько то
Жизней - столько то
Потрачено золото столько то

**Comment by dosinabox** [comment]
![Screenshot_1](https://user-images.githubusercontent.com/39790055/75806572-f67ab000-5d94-11ea-8f7d-f5838f2fd9a2.jpg)

> Потрачено жизней

Нужно ли учитывать траты на улучшение Когтя?

**Comment by Kor-Angar** [comment]
Если только отдельно сделать. И написать что на улучшение когтя потрачено столько то жизней.

**Comment by Kor-Angar** [comment]
Тогда и кольца добавить. Сколько  Лорду Хагену отдано 😀И альманах которые Пирокару сдаются 😀

**Comment by dosinabox** [comment]
точно!

**Comment by Kor-Angar** [comment]
а "полковники" орков считаются как предводители орков?

**Comment by dosinabox** [comment]
![Screenshot_1](https://user-images.githubusercontent.com/39790055/75814998-b838bd00-5da3-11ea-89ed-bfedf88d0dc9.jpg)

> а "полковники" орков считаются как предводители орков?

Да. aivar[AIV_MM_REAL_ID] = ID_ORCCOMMANDER.

**Comment by Kor-Angar** [comment]
там еще может быть Маг с рунами Белиара

**Comment by dosinabox** [comment]
> там еще может быть Маг с рунами Белиара

Не суть, просто максимальный уровень будет 5 или 6.

**Comment by Kor-Angar** [comment]
а можно сделать отдельную книгу по параметрам?

***


Левая страница | Правая страница
-- | --
Сила 135 | 1р  75%
45 поднято за 65 LP | 45 поднято за 65 LP
1 бонус Эрола | 2 бонус Вульфгара
10 молитвы | 5 книга
12 скрижали | 12 скрижали
2 яблоки |  
2 суп Теклы |  
1 Пламенная нарезка |  
1 Мясная похлёбка | 2р = 75%
3 Драконий Корень | 45 поднято за 65 LP
10 Эликсиры Силы | 2 бонус Сержио
3 Драконий напиток |  
5 слезы Инноса |  
30 Embarla Firgasto |  

***
или

Левая страница | Правая страница
-- | --
Глава: 4 | 135 силы
Гильдия: ОНД | 45 поднято за 65 LP
Игнор Бонусов: Да | 1 бонус Эрола
  | 10 молитвы
1р = 75% | 12 скрижали
45 поднято за 65 LP | 2 яблоки
2 бонус Вульфгара | 2 суп Теклы
5 книга | 1 Пламенная нарезка
12 скрижали | 1 Мясная похлёбка
  | 3 Драконий Корень
2р = 75% | 10 Эликсиры Силы
45 поднято за 65 LP | 3 Драконий напиток
2 бонус Сержио | 5 слезы Инноса
  | 0 Embarla Firgasto

***



возможно общественность в конце игры будет выкладывать скрины :-))

**Comment by dosinabox** [comment]
![Screenshot_1](https://user-images.githubusercontent.com/39790055/77319186-6f7b8080-6d1f-11ea-8119-d0977ab50329.jpg)

- информация об альманахах/яйцах/кольцах показывается в зависимости от гильдии ГГ;
- учитываются грибы, гоблинские ягоды и драконьи корни, сваренные в зельях;
- добавлена информация о сборке обновления, testamulet теперь не нужен и удален;
- добавлена информация об уничтожении Когтя.

**Comment by Kor-Angar** [comment]
> учитываются грибы, гоблинские ягоды и драконьи корни, сваренные в зельях;

а как же понять тогда, сколько съел а сколько сварил? :-)

**Comment by Kor-Angar** [comment]
в убитые можно добавить обычных орков

**Comment by Kor-Angar** [comment]
можно так же учитывать царский щавель сваренный в зельях. это наверное важнее.

**Comment by dosinabox** [comment]
> а как же понять тогда, сколько съел а сколько сварил? :-)

Можно попробовать их разделить в тексте, но не учитывать их совсем тоже неправильно.

**Comment by Kor-Angar** [comment]
> Можно попробовать их разделить в тексте, но не учитывать их совсем тоже неправильно.

можно считать сколько съел герой ягод/корней и считать сколько ушло царских щавелей на напитки. и тогда будет норм.
например я потратил 10 царских на эликсиры и съел 3 ягоды. значит ягод всего было 10+3=13.

**Comment by Kor-Angar** [comment]
а нельзя статистику по алхимии просто вынести в журнал в раздел по алхимии?

**Comment by dosinabox** [comment]
> например я потратил 10 царских на эликсиры и съел 3 ягоды. значит ягод всего было 10+3=13.

Эликсиры могут быть не только из ягод)

**Comment by Kor-Angar** [comment]
могут. но для того что бы полностью учесть нужен отдельный блок по алхимии. а его логичнее делать в имеющемся разделе дневника ;-)

**Comment by dosinabox** [comment]
- добавлена статистика о мастерах;
- добавлена информация о царских щавелях.

![Screenshot_1](https://user-images.githubusercontent.com/39790055/77472340-cd45c080-6e24-11ea-9827-95b9ab32b712.jpg)

**Comment by dosinabox** [comment]
statsbook в Strange Brigade :)

![312670_screenshots_20200719225240_1](https://user-images.githubusercontent.com/39790055/89260799-a8185880-d635-11ea-895b-fc047696d841.jpg)

**Comment by Kor-Angar** [comment]
может сделать что бы книга вставлялась в инвентарь через настройку неофициально обновления у помощника?

**Comment by dosinabox** [comment]
Должны ли убийства от рук соратников и призванных существ увеличивать счетчик убийств в книге? Или оставить только убийства **лично** от рук ГГ?

> может сделать что бы книга вставлялась в инвентарь через настройку неофициально обновления у помощника?

Можно, но не вижу смысла.

**Comment by Kor-Angar** [comment]
Мне кажется что нет. Или можно выводить две цифры. Сколько убил ГГ и сколько всего убиты.

**Comment by Kor-Angar** [comment]
в статистике у Боспера не учитываются первые шесть шкур волка.
![image](https://user-images.githubusercontent.com/49514764/93161544-70fc9300-f71b-11ea-9bf9-6f6a8ba79680.png)
![image](https://user-images.githubusercontent.com/49514764/93161551-74901a00-f71b-11ea-8d94-3ab650e8b0f1.png)

**Comment by dosinabox** [comment]
исправлено

**Comment by dosinabox** [comment]
- добавлена информация о полученном опыте от краж;
- добавлена информация о количестве украденного золота.

**Comment by Kor-Angar** [comment]
> добавлена информация о полученном опыте от краж;

всё таки я дождался :-) жаль что не в начале года :-)

**Comment by Kor-Angar** [comment]
можно так же добавит болотную траву для Фортуно
![image](https://user-images.githubusercontent.com/49514764/106503636-f7512000-64d6-11eb-9963-bd3de710b70c.png)

**Comment by Kor-Angar** [comment]
а в книги есть проверка на фракцию? 

- Наём/ ОнД - яйца и Рога
- Пал - кольца орков
- Маг - альманахи

***
Главу наверное еще надо добавить ;-)

**Comment by Kor-Angar** [comment]
на мой взгляд предводителей орков можно ОнД/магу не показывать, так как он всего один :-)
![20210228202430_1](https://user-images.githubusercontent.com/49514764/109427400-49686100-7a03-11eb-9488-53520417aacd.jpg)

проверка есть на то отдал ли ГГ коготь?

**Comment by Kor-Angar** [comment]
информации о том ли расширена долины нету.

**Comment by dosinabox** [comment]
> а в книги есть проверка на фракцию?

да

> проверка есть на то отдал ли ГГ коготь?

да

> информации о том ли расширена долины нету.

это видно и без книги

> предводителей орков можно ОнД/магу не показывать, так как он всего один :-)

убит ли этот один орк - тоже статистика)

**Comment by Kor-Angar** [comment]
> это видно и без книги

а как понять из скрина книги со статистикой, что это статистика без расширенного миненталя? ;-)

**Comment by dosinabox** [comment]
во-первых, определить расширенный Миненталь или нет - можно только после его непосредственной загрузки
во-вторых, после загрузки расширенного Миненталя игрок может удалить файл с ним, что сделает его обычным

```cpp
func int C_OldWorldIsWasteland()
{
	if(Mob_HasItems("D36_SECRET_WASTELAND_CHEST",ItPl_Mushroom_02))
	{
		return TRUE;
	};
	return FALSE;
};
```

**Comment by Kor-Angar** [comment]
ну понятно, что может удалить. понятно что понять можно после загрузки. я просто это предлагаю потому, что в Минентале кол-во орков увеличено и доп Мракорисы есть. и грубо говоря если я вижу чужой скрин статистики и вижу там другое кол-во рогов мракорисов (оговорка, я рога не собираю, но есть граждане которые ну очень рьяно к ним относится) то начинают возникать вопросы. а надпись что это расширенная версия миненталя, например внизу справа сразу же эти вопросы снимает ;-) то же самое и с элитными орками :-) это с одной стороны, с другой стороны это же надпись будет вызывать интерес у тех у кого по какой то причине этой локации нет.

а счётчик обычных орков влезет?
те орки на ирдорате которые окружают полковника считаются как элитные или как предводители?

**Comment by dosinabox** [comment]
> а счётчик обычных орков влезет?

да

> те орки на ирдорате которые окружают полковника считаются как элитные или как предводители?

элитные

> надпись что это расширенная версия миненталя

Добавлено - запись появится в книге, если первая загрузка Миненталя была с файлом Data/ModVDF/g2a_nr_scriptpatch_v2*_wasteland.mod.
Если файл был установлен после первой загрузки, то записи не будет. Новых монстров тоже не будет (заселение мира уже состоялось без них).
Если файл был удален после первой загрузки, то запись остается. Заселенные монстры остаются в сохранении. А рельеф местности может и не остается)

**Comment by Kor-Angar** [comment]
Если место есть тогда можно и обычных орков добавить, самых массовых так сказать 😀Главу, фракцию есть куда впихнуть? 😀23:21, 1 марта 2021 г., dosinabox <notifications@github.com>:

а счётчик обычных орков влезет?

да

те орки на ирдорате которые окружают полковника считаются как элитные или как предводители?

элитные

надпись что это расширенная версия миненталя

Добавлено - запись появится в книге, если первая загрузка Миненталя была с файлом g2a_nr_scriptpatch_v2_wasteland.mod*.
Если файл был установлен после первой загрузки, то записи не будет. Новых монстров тоже не будет (заселение мира уже состоялось без них).
Если файл был удален после первой загрузки, то запись остается. Заселенные монстры остаются в сохранении. А рельеф местности может и не остается)

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.
-- Отправлено из мобильного приложения Яндекс.Почты

**Comment by dosinabox** [comment]
> обычных орков добавить, самых массовых так сказать

Кто должен учитываться в число убитых орков?

- [x] орки-разведчики
- [x] орки-воины
- [ ] орки-шаманы
- [ ] элитные орки
- [ ] полковники орков
- [ ] предводители орков
- [ ] орки-нежить
- [ ] Хош-Пак
- [ ] Ур-Шак
- [ ] главарь орды

> Главу, фракцию есть куда впихнуть?

Зачем добавлять туда фракцию, если она показывается на экране статов персонажа?

**Comment by Kor-Angar** [comment]
>  Кто должен учитываться в число убитых орков?

орки-разведчики и  орки-воины

> Зачем добавлять туда фракцию, если она показывается на экране статов персонажа?

что бы картинка была целостная. например у ОнД больше людей-ящеров , а у Паладина больше орков, а у Мага больше Ищущих.
смотришь на скрин и понятно что это паладин он убил столько то орков. или смотришь на скрин и не понятно, так как ОнД убил слишком много Ищущих :-)


>  Таррок

это кто? ур-шак? :-)

**Comment by dosinabox** [comment]
> что бы картинка была целостная. например у ОнД больше людей-ящеров , а у Паладина больше орков, а у Мага больше Ищущих.

Гильдию можно узнать по разделу "Отдано" (ОнД - рога и яйца, паладин - кольца, маг - альманахи).

> это кто? ур-шак? :-)

Да, точно.

**Comment by Kor-Angar** [comment]
главарь орды это шаман из Яркендара?

**Comment by Kor-Angar** [comment]
кажется так

### Орк-Воин
Орк-Воин / разведчик (хз с чем)
Орк-Воин (средний топор)
Орк-Воин (тяжелый топор)
### Орк-шаман
Орк-шаман (легкий топор)
Орк-шаман / Ур-Шак (хз с чем)
Орк-шаман / Хош-Пак (хз с чем)
Орк-шаман / главарь орды (хз с чем)
### Элитный Орк
Элитный Орк (Брутальный топор)
Элитный Орк / предводители орков (Брутальный топор)
Элитный Орк (Меч)
Элитный Орк / полковник орков (Брутальный топор)
Элитный Орк / полковник орков (топор 140/140)
### Орки-нежить
Орк-нежить

**Comment by Kor-Angar** [comment]
> Гильдию можно узнать по разделу "Отдано" (ОнД - рога и яйца, паладин - кольца, маг - альманахи).

из книги статистики много что можно узнать. например понять что это Ночь Война, определить фракцию, даже главу прикинуть. и понять например что если отданы мечи то мастер Гарад, а если Шкуры то мастер Боспер :-) а если Грибы то мастер Костя. // я знаю что они подписаны, но мысль развиваю //
но с точки зрения информативности правильнее писать и название полностью Готика 2 НВ ( или классик)  и фракция (нейтрал, наём, ОнД) и главу(2,3). иначе не все  информацию расшифруют, а многие и расшифруют неправильно, будут задавать вопросы, строить теории и даже обзываться :-) 
вот мне тут эпос прислали на днях в Вк, что я бла бла бла, а играть надо в АБ... бла, бла... а книгу статистика твою оказывается даёт Пирокар, после сдачи Альманаха  бла бла бла

**Comment by dosinabox** [comment]
Написать можно много чего, и я совсем не против, но вот места уже почти не осталось. Может, в будущем можно будет листать страницы, как сделано в некоторых модах. Но я так делать не умею.

**Comment by Kor-Angar** [comment]
Фракцию и главу можно запихать в ту строчку где убито.

а название игры Готика 2 НВ / Готика 2 классик вместо "информация о сборке"

**Comment by Kor-Angar** [comment]
> Написать можно много чего, и я совсем не против, но вот места уже почти не осталось. Может, в будущем можно будет листать страницы, как сделано в некоторых модах. Но я так делать не умею.

а почему просто не сделать через обычный диалог?

https://www.youtube.com/watch?v=eZSxV-wjZpE

**Comment by dosinabox** [comment]
Интересно) Да, хорошая реализация, можно сделать так.

**Reply by Kor-Angar** [reply]
у меня код черновой. уверен что ты сделаешь более качественно. но на всякий случай вот :-) 

```D
//****************************************************************************
//				ItWr_SuperBook
//			---------------------------------------------
//****************************************************************************

var int SuperBook;

var string text_1;
var string text_2;
var string text_3;
var string text_4;
var string text_5;

var int nDocID;

func void text_12345()
{	
	Doc_SetPages(nDocID,2);
		
	Doc_SetPage(nDocID,0,"Book_Wood_L.tga",0);
	Doc_SetPage(nDocID,1,"Book_Wood_R.tga",0);
	Doc_SetMargins(nDocID,0,270,20,20,20,1);
	Doc_SetMargins(nDocID,1,20,20,270,20,1);
	font_10 = "font_10_book.TGA";
	font_15 = "font_15_book.TGA";
	
	// Первая страница.	
	Doc_SetFont(nDocID,-1,font_15);//Doc_SetFont(nDocID,-1,"font_15_book.tga");
	//Doc_SetMargins(nDocID,0,660,180,40,180,1);
	Doc_PrintLine(nDocID,0,"");
	Doc_PrintLine(nDocID,0,"Заметки Kor Angara");
	Doc_PrintLine(nDocID,0,"-----------");
	Doc_SetFont(nDocID,-1,font_10);//Doc_SetFont(nDocID,-1,"font_10_book.TGA");
	Doc_PrintLine(nDocID,0,"");
	Doc_PrintLines(nDocID,0,text_1);
	Doc_PrintLine(nDocID,0,"");
	Doc_PrintLines(nDocID,0,text_2);
	Doc_PrintLine(nDocID,0,"");
	Doc_PrintLines(nDocID,0,text_3);
	
	// Вторая страница.
	Doc_PrintLine(nDocID,1,"");
	Doc_PrintLine(nDocID,1,"");
	Doc_PrintLine(nDocID,1,"");
	Doc_PrintLines(nDocID,1,text_4);
	Doc_PrintLine(nDocID,1,"");
	Doc_PrintLines(nDocID,1,text_5);
	
	Doc_Show(nDocID);
};


func void text_12345_1()
{
	text_1 = "         Том первый.";
	text_2 = "111111 111111";
	text_3 = "222222 222222";
	text_4 = "333333 333333";
	text_5 = "444444 444444";
	text_12345();
};

func void text_12345_2()
{
	text_1 = "         Том второй.";
	text_2 = "111111 111111";
	text_3 = "222222 222222";
	text_4 = "333333 333333";
	text_5 = "444444 444444";
	text_12345();
};

func void text_12345_3()
{
	text_1 = "         Том 3.";
	text_2 = "111111 111111";
	text_3 = "222222 222222";
	text_4 = "333333 333333";
	text_5 = "444444 444444";
	text_12345();
};

func void text_12345_4()
{
	text_1 = "         Том 4.";
	text_2 = "111111 111111";
	text_3 = "222222 222222";
	text_4 = "333333 333333";
	text_5 = "444444 444444";
	text_12345();
};

func void text_12345_0()
{
	nDocID = Doc_Create();
	AI_ProcessInfos(self);
	text_12345_1();
};

instance ItWr_SuperBook(Worldmap)
{
	on_state[0] = Use_SuperBook;
};
func void Use_SuperBook()
{
	SuperBook	= true;
	text_12345_0();		// 
};

prototype PC_SuperBook(C_Info)
{
	npc = PC_Hero;
	nr = 999;
	condition = PC_SuperBook_Condition;
	important = 0;
	permanent = 1;
};

instance PC_SuperBook_1(PC_SuperBook)
{
	information = text_12345_1;
	description = "text_12345_1";
};
instance PC_SuperBook_2(PC_SuperBook)
{
	information = text_12345_2;
	description = "text_12345_2";
};
instance PC_SuperBook_3(PC_SuperBook)
{
	information = text_12345_3;
	description = "text_12345_3";
};
instance PC_SuperBook_4(PC_SuperBook)
{
	information = text_12345_4;
	description = "text_12345_4";
};
instance PC_SuperexitBook(PC_SuperBook)
{
	information = exitBook;
	description = "конец.";
};

func int PC_SuperBook_Condition()
{
	if (SuperBook == TRUE)
	{
		return TRUE;
	};	
};

func void exitBook()
{
	AI_StopProcessInfos(self);
};
```

**Comment by Kor-Angar** [comment]
в статистики нет информации об отданной болотной траве Фортуно.

**Comment by dosinabox** [comment]
> в статистики нет информации об отданной болотной траве Фортуно.

добавлено

**Comment by Kor-Angar** [comment]
конвертируй пожалуйста в дискуссию

**Reply by Kor-Angar** [reply]
спасибо

**Comment by Kor-Angar** [comment]
проверь пожалуйста если будет возможность корректно ли считается 

- [ ] атрибут поднятый у учителя
- [ ] траты ЛП у учителей


книгу статистики даёт СториХелпер 
 https://steamcommunity.com/sharedfiles/filedetails/?id=2799388566

**Reply by Kor-Angar** [reply]
вроде теперь все бонусы ловкости считаются корректно и в голд и неоф. обновление.
для теста добавил при открывание книги статистики в разделе ловкости добавление герою в инвентарь перманентных зелий, ягод, табличек и изучение языков. бонус ренгару также учитывается.

**Reply by Kor-Angar** [reply]
считает и B_RaiseFightTalent из голд и ChangeTalent_Hero в неоф. патче корректно с настройкой "честный расчёт стоимости"
![20220426015024_1](https://user-images.githubusercontent.com/49514764/165187524-121bafb8-6030-4f24-99e5-2420d2631601.jpg)
![20220426015051_1](https://user-images.githubusercontent.com/49514764/165187528-242a8427-0ced-4f92-be87-3ad8c32d7c0b.jpg)

**Reply by Kor-Angar** [reply]
обновил в стиме

**Reply by Kor-Angar** [reply]
@dosinabox  напиши плиз, запускается ли у тебя с этим мини-модом. если игра запускается с неоф. патчем. я доделаю страничку про силу и выложу в паблик.

**Reply by dosinabox** [reply]
работает
![20220502215314_1](https://user-images.githubusercontent.com/39790055/166315971-670f0872-cbe9-4068-8e3a-7153595b8a45.jpg)

**Comment by Kor-Angar** [comment]
визуально сделал вот так:
слева - атрибут, справа навыки. так же справа на верху расчёт урона  одетого оружия.

на картинки там где по русски - неоф. патч. там где не по русски голд версия
![20220426014012_1](https://user-images.githubusercontent.com/49514764/165187837-0c3db487-bb0e-43c7-aec9-731d026e5d4c.jpg)
![20220426015051_1](https://user-images.githubusercontent.com/49514764/165187830-dc6302fe-a6f6-4c06-a6c6-bee163108a71.jpg)

если есть предложения по надписям буду рад :-)

**Reply by Kor-Angar** [reply]
финал где то близко :-)
![20220521004332_1](https://user-images.githubusercontent.com/49514764/169616613-53f02a71-21c3-4ad1-9a63-98888d458279.jpg)

**Comment by dosinabox** [comment]
Следует ли сбрасывать статистику о бонусах от еды и молитв при полном сбросе через помощника?
```cpp
Stats_Blessings_GoldGiven = 0;
Stats_Blessings_Str = 0;
Stats_Blessings_Dex = 0;
Stats_Blessings_MaxHp = 0;
Stats_Blessings_MaxMana = 0;
Stats_Beliar_ClawMaxHp = 0;
Stats_Beliar_HpGiven = 0;
Stats_Beliar_ManaGiven = 0;
Stats_Beliar_GoldTaken = 0;
TotalDexEaten = 0;
TotalStrEaten = 0;
TotalApplesEaten = 0;
TotalMushroomsEaten = 0;
```

**Reply by Kor-Angar** [reply]
Если это используется для тестов, то стоит.
если для игры, то нет. Можно там же сделать сброс опциональным, захотел сбросил, захотел оставил.

**Comment by Kor-Angar** [comment]
можно в книги статистики вывести ник из стима `Hlp_GetSteamPersonalName()`

![image](https://user-images.githubusercontent.com/49514764/200895590-53567af8-11ba-4d08-ac85-df06c5edefb9.png)
